### ISSUE TITLE
When there is no camera available in the system, an exception occurs.

### ISSUE THREAD

**Issue by zhouhaooo** [kcwongjoe/directshow_camera#10]
An exception is raised when I use "std::vector<CameraDevice> devices = camera.getCameras();" to retrieve the list of devices. 
The error stems from the checkCreateClassEnumeratorResult function failing to handle the HRESULT hr correctly.
When hr == S_FALSE, SUCCEEDED(hr) returns true, which is not the expected behavior.

```
bool checkCreateClassEnumeratorResult(HRESULT hr, std::string* errorString, std::string errorDescription)
{
    bool result = true;
    if ( hr!= S_OK )
    {
        result = false;

        // Pass error string
        if (errorString != NULL)
        {
            *errorString = errorDescription + ": ";
            if (hr == E_OUTOFMEMORY)
            {
                *errorString += "There is not enough memory available to create a class enumerator.";
            }
            else if (hr == S_FALSE)   // Cannot be reached when hr == S_FALSE.
            {
                *errorString += "There is no any camera.";
            }
            else if (hr == E_POINTER)
            {
                *errorString += "NULL pointer argument.";
            }
            else
            {
                *errorString += "hr = (" + std::to_string(hr) + ")";
            }
        }
    }

    return result;
}
```

**Comment by kcwongjoe** [comment]
Thanks. I am restructuring the whole project. Will handle this situation on the new version as well.

**Comment by kcwongjoe** [comment]
Thanks.  Has been fixed on 020f02be288eac2732d928a88e1cfcb24f0cad76.

**Comment by kcwongjoe** [comment]
Thanks. has been fixed on v2
